spec/build_code_calculator_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe Fastlane::Wpmreleasetoolkit::Versioning::BuildCodeCalculator do
+  describe 'calculates the next build code' do
+    it 'increments the simple build code by 1' do
+      build_code = 735
+      bumped_build_code = described_class.new.next_simple_build_code(after: build_code)
```

Comment:
As I mentioned in my comment above, according to the YARD documentation you wrote for `next_simple_build_code`, it's documented to expect a `BuildCode` instance, not an `Integer` like `735` here.

To fix this you should either rewrite the unit tests to make the `build_code` value passed as parameter be a `BuildCode.new(735)` — which will then require to fix your implementation of `next_simple_build_code` as discussed in the other comment — or we could completely remove the `BuildCode` model class and directly use `Integer` instances everywhere we currently had `BuildCode` instances, to make the call site more like the one you use here in this test, which might be even simpler after all.